lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -3518,15 +3518,30 @@ private long prepareCommitInternal() throws IOException {
       }
 
       if (pointInTimeMerges != null) {
+        MergePolicy.OneMerge nextMerge = null;
+
+        if (pendingMerges.size() > 0) {
+          // nocommit getting OneMerge instance here via mergeSource.getNextMerge() will
```

Comment:
How about passing the `pendingMerges` to the event listener instead?  (Sorry if I asked for `OneMerge` on the issue!  `MergeSpecification` is better since it can hold multiple merges, allows event listener to log things like how many merges were requested during `commit`, etc.).

Reply:
No problem! From the context I assume you are actually meaning to pass `pointInTimeMerges` into event listener, since `pendingMerges` is of type `Deque<MergePolicy.OneMerge>`? I've pushed a commit to use `pointInTimeMerges` for now.